### PR TITLE
Nan property handling

### PR DESCRIPTION
**Pull request by domportera**
this took way longer to find than i'm proud to admit

for #123 

and apparently #177

### CODE REVIEW

**Comment by pixtur** [comment]
Thanks, this is already a great fix! 

One thing we could improve in the long term is to actually use the defined default value of the edited property.

This might be possible by returning 

```
    public enum InputEditStateFlags
    {
        Nothing = 0x0,
        Started = 0x1,
        Modified = 0x2,
        Finished = 0x4,
        ShowOptions = 0x8,
        ResetToDefault = 0x10,
        ModifiedAndFinished = Modified | Finished
    }
```
`InputEditStateFlags.ResetToDefault`  

I will quickly look into this.

**Comment by domportera** [comment]
Thanks! It took a few iterations to find the real *root* of the problem but I'm glad I found it.

I think I tried ResetToDefault and didn't have luck, but im sure I was doing something wrong. I'm sure you'll have better luck than I did :)

File: T3/Gui/Interaction/SingleValueEdit.cs [hunk 1]
```diff
@@ -390,6 +401,7 @@ private enum InputStates
 
         private static string _numberFormat = "{0:0.000}";
         private static double _timeOpened;
+        private const double DefaultValue = 0.0;
```
Comment (pixtur) [inline T3/Gui/Interaction/SingleValueEdit.cs:404]: should be `_defaultValue` because of private

**Comment by pixtur** [comment]
You code looks really nice and works. I'm really impressed how quickly you were able to dive into that.

File: T3/Gui/Interaction/SingleValueEdit.cs (hunk 1 above)
Comment (domportera) [reply T3/Gui/Interaction/SingleValueEdit.cs:404]: noted!

**Comment by domportera** [comment]
thanks!! i do my best :) just took a lot of hunting and a strong magnifying glass 🕵️‍♂️

**Pull request merged**